src/wallet/walletdb.cpp

```diff
@@ -452,7 +447,12 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
         }
         else if (strType == "defaultkey")
         {
-            ssValue >> pwallet->vchDefaultKey;
+            CPubKey vchPubKey;
+            ssKey >> vchPubKey;
```

Comment:
This should be `ssValue >> vchPubKey`. Otherwise this will fail for every wallet with a default key record!